lib/Fhp/Protocol/Message.php

```diff
@@ -283,11 +283,13 @@ public static function parse($rawMessage)
             // Signature header and footer must always be there when the "encrypted" structure was used.
             $result->signatureHeader = array_shift($result->plainSegments);
             $result->signatureFooter = array_pop($result->plainSegments);
-            if (!($result->signatureHeader instanceof HNSHKv4)) {
-                throw new \InvalidArgumentException("Expected first segment to be HNSHK: $rawMessage");
-            }
-            if (!($result->signatureFooter instanceof HNSHAv2)) {
-                throw new \InvalidArgumentException("Expected last segment to be HNSHA: $rawMessage");
+
+            $signatureHeaderAsExpected = $result->signatureHeader instanceof HNSHKv4;
```

Comment:
Das funktioniert so nicht. Wir wollen nicht die _Typprüfung_ optional machen, sondern die Frage ob das Segment überhaupt da ist. Wenn es _nicht_ da ist, dann hätte `array_shift()` oben ein x-beliebiges anderes Segment entfernt, was uns dann in der Antwort fehlt. Wenn es aber da sein sollte, dann wollen wir auch, dass es v4/v2 ist, denn andere Versionen werden derzeit nicht unterstützt (dafür gibt es eine `UnsupportedException`, die ich noch hochladen muss).

Reply:
Der Kommentar in Zeile 283 "Signature header and footer must always be there when the "encrypted" structure was used." trifft schon nicht mehr zu, also ab da muss geändert werden. Vermutlich ein `if` so ähnlich wie das in Zeile 276, und dann ein `else` so ähnlich wie in Zeile 296.

Reply:
Ah da hab ich geschlafen  😓, bist du mit #122 zufrieden?